permission_handler/ios/Classes/strategies/CriticalAlertsPermissionStrategy.m

```diff
@@ -0,0 +1,77 @@
+//
+//  CriticalAlertsPermissionStrategy.m
+//  permission_handler
+//
+//  Created by Neal Soni on 2021/6/8.
+//
+
+#import "CriticalAlertsPermissionStrategy.h"
+
+#if PERMISSION_CRITICAL_ALERTS
+
+@implementation CriticalAlertsPermissionStrategy
+
+- (PermissionStatus)checkPermissionStatus:(PermissionGroup)permission {
+  return [CriticalAlertsPermissionStrategy permissionStatus];
+}
+
+- (ServiceStatus)checkServiceStatus:(PermissionGroup)permission {
+  return ServiceStatusNotApplicable;
+}
+
+- (void)requestPermission:(PermissionGroup)permission completionHandler:(PermissionStatusHandler)completionHandler {
+  PermissionStatus status = [self checkPermissionStatus:permission];
+  if (status != PermissionStatusDenied) {
+    completionHandler(status);
+    return;
+  }
+  dispatch_async(dispatch_get_main_queue(), ^{
+    if(@available(iOS 12.0, *)) {
+      UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+      UNAuthorizationOptions authorizationOptions = 0;
+      authorizationOptions += UNAuthorizationOptionCriticalAlert;
+      [center requestAuthorizationWithOptions:(authorizationOptions) completionHandler:^(BOOL granted, NSError * _Nullable error) {
+        if (error != nil || !granted) {
+          completionHandler(PermissionStatusPermanentlyDenied);
+          return;
+        }
+
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Do we specifically need to dispatch the `[[UIApplication sharedApplication] registerForRemoteNotifications]` on the main queue here?

I am aware that the `completionHandler` needs to be ran on the main queue, but I think we should solve this problem in the implementation of the `completionHandler` block instead of all the places where it is used (this is a design error on my side which this PR made me aware of, so thanks for that ;) ).

Reply:
I'm not sure if this fallback is even needed — Just kept it since it was in the `NotificationPermissionStrategy.m`

Reply:
Clear, I see this is a bigger problem through out the code base. I will accept this and sweep through all strategies and make sure the `dispatch_async(dispatch_get_main_queue(), ^{  })` is done on the correct place.

Reply:
Thank you! Sorry sorry, I'm not an objective-c expert